src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -193,6 +204,7 @@ private async Task RunStepsAsync(List<IStep> embeddedSteps)
 
                 await RunStepAsync(step);
 
+                // TODO: What if a composite action is not skipped, but all of its steps are skipped, do we mark it as skipped?
```

Comment:
We should think about this a bit

We are leaning on composite actions to be the way to share steps. So consider a composite action that contains a a mix of nested steps: some run always; some do not.

Let's identify what we want long-term. Worried about changing behavior and breaking customers. Might want to hold off on nested `if` support altogether for now.